### PR TITLE
rustc: Rename rustc_macro to proc_macro

### PR DESCRIPTION
**Pull request by alexcrichton**
This commit blanket renames the `rustc_macro` infrastructure to `proc_macro`,
which reflects the general consensus of #35900. A follow up PR to Cargo will be
required to purge the `rustc-macro` name as well.

### CODE REVIEW

**Comment by alexcrichton** [comment]
r? @nrc 

cc @dtolnay, @sgrif 

This is a [breaking-change] from the current implementation, just blindly renaming everything. I'm curious what you think, though, about this strategy. Do you think it's worth implementing the infrastructure to ensure a smooth transition? It unfortunately wouldn't be easy but it also wouldn't be impossible.

Also one point I forgot to mention was that the existing `proc_macro` has been blanket renamed to `proc_macro_plugin`, but I think that most of it can move to `proc_macro`. I figured that could be left to a future PR, however.

**Comment by dtolnay** [comment]
I think this is fine without infrastructure to ensure a smooth transition. We'll just fix everything when the nightly comes out, same as usual.

**Comment by sgrif** [comment]
Yeah I'm pretty sure this breaks diesel, serde, and maybe 1-2 other crates. We'll be fine. Just :shipit: :smile:

File: src/libproc_macro_plugin/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+[package]
+authors = ["The Rust Project Developers"]
+name = "proc_macro_plugin"
+version = "0.0.0"
```
Comment (sgrif) [inline src/libproc_macro_plugin/Cargo.toml]: Way better than 1.0

**Comment by Mark-Simulacrum** [comment]
Travis failed:
```
error[E0061]: this function takes 0 parameters but 1 parameter was supplied
   --> src/libserialize/json.rs:436:26
    |
436 |     escape_str(writer, v.encode_utf8(&mut [0; 4]))
    |                          ^^^^^^^^^^^ expected 0 parameters

error[E0308]: mismatched types
   --> src/libserialize/json.rs:436:24
    |
436 |     escape_str(writer, v.encode_utf8(&mut [0; 4]))
    |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^ expected &str, found struct `rustc_unicode::char::EncodeUtf8`
    |
    = note: expected type `&str`
    = note:    found type `rustc_unicode::char::EncodeUtf8`

error: aborting due to 2 previous errors
```

**Comment by alexcrichton** [comment]
@bors: retry force clean

* restarted buildbot

**Comment by alexcrichton** [comment]
@bors: r-

That's legit I think, gotta update Cargo

**Pull request merged**